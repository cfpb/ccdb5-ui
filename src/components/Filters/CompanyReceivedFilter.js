/* eslint complexity: ["error", 7] */
import './DateFilter.less';
import { DATE_VALIDATION_FORMAT, maxDate, minDate } from '../../constants';
import React, { useEffect, useMemo, useRef, useState } from 'react';
import {
  selectQueryCompanyReceivedMax,
  selectQueryCompanyReceivedMin,
} from '../../reducers/query/selectors';
import { useDispatch, useSelector } from 'react-redux';
import { changeDates } from '../../actions/filter';
import CollapsibleFilter from './CollapsibleFilter';
import dayjs from 'dayjs';
import dayjsCustomParseFormat from 'dayjs/plugin/customParseFormat';
import dayjsIsBetween from 'dayjs/plugin/isBetween';
import { formatDate } from '../../utils/formatDate';
import iconMap from '../iconMap';

dayjs.extend(dayjsCustomParseFormat);
dayjs.extend(dayjsIsBetween);

export const CompanyReceivedFilter = () => {
  const fieldName = 'company_received';
  const title = 'The date the CFPB sent the complaint to the company';
  const dateFrom = useSelector(selectQueryCompanyReceivedMin);
  const dateThrough = useSelector(selectQueryCompanyReceivedMax);

  const initialFromDate = dayjs(dateFrom).isValid() ? formatDate(dateFrom) : '';
  const initialThroughDate = dayjs(dateThrough).isValid()
    ? formatDate(dateThrough)
    : '';

  const [fromDate, setFromDate] = useState(initialFromDate);
  const [throughDate, setThroughDate] = useState(initialThroughDate);
  const dispatch = useDispatch();
  const errorMessageText = "'From' date must be less than 'through' date";

  const fromRef = useRef();
  const throughRef = useRef();

  useEffect(() => {
    // put it in YYYY-MM-DD format
    // validate to make sure it's not invalid
    const validFromDate = dateFrom ? formatDate(dateFrom) : '';
    setFromDate(validFromDate);
  }, [dateFrom]);

  useEffect(() => {
    const validThroughDate = dateThrough ? formatDate(dateThrough) : '';
    setThroughDate(validThroughDate);
  }, [dateThrough]);

  const handleClear = (period) => {
    if (period === 'from') {
      dispatch(changeDates(fieldName, '', throughDate));
    }
    if (period === 'through') {
      dispatch(changeDates(fieldName, fromDate, ''));
    }
  };

  const handleKeyDownFromDate = (event) => {
    if (event.key === 'Enter') {
      fromRef.current.blur();
    }
  };

  const handleKeyDownThroughDate = (event) => {
    if (event.key === 'Enter') {
      throughRef.current.blur();
    }
  };

  const errors = useMemo(() => {
    const errs = [];
    if (dayjs(fromDate).isAfter(throughDate)) {
      errs.push(errorMessageText);
    }
    if (dayjs(fromDate).isBefore(minDate)) {
      errs.push(
        "'From' date must be after " +
          dayjs(minDate).format(DATE_VALIDATION_FORMAT)
      );
    }
    if (dayjs(throughDate).isAfter(maxDate)) {
      errs.push(
        "'Through' date must be before " +
          dayjs(maxDate).format(DATE_VALIDATION_FORMAT)
      );
    }

    return errs;
  }, [fromDate, throughDate]);

  const handleDateChange = () => {
    let _throughDate = throughDate;
    let _fromDate = fromDate;
    if (_fromDate && !dayjs(fromDate).isValid()) {
      fromRef.current.value = '';
      _fromDate = '';
    }
    if (_throughDate && !dayjs(throughDate).isValid()) {
      throughRef.current.value = '';
      _throughDate = '';
    }
    const isDateDifferent =
      dateFrom !== _fromDate || dateThrough !== _throughDate;
    if (isDateDifferent) {
      dispatch(changeDates(fieldName, _fromDate, _throughDate));
    }
  };

  const inputFromClassName = useMemo(() => {
    const style = ['a-text-input'];
    if (
      dayjs(fromDate).isBefore(minDate) ||
      dayjs(fromDate).isAfter(throughDate)
    ) {
      style.push('a-text-input__error');
    }
    return style.join(' ');
  }, [fromDate, throughDate]);

  const inputThroughClassName = useMemo(() => {
    const style = ['a-text-input'];
    if (
      dayjs(throughDate).isAfter(maxDate) ||
      dayjs(throughDate).isBefore(fromDate)
    ) {
      style.push('a-text-input__error');
    }
    return style.join(' ');
  }, [fromDate, throughDate]);

  return (
    <CollapsibleFilter title={title} className="aggregation date-filter">
      <div>
        <ul className="date-inputs">
          <li>
            <label
              className="a-label a-label__heading body-copy"
              htmlFor={`${fieldName}-from`}
            >
              From
            </label>
            <div className="m-btn-inside-input">
              <input
                id={`${fieldName}-from`}
                className={inputFromClassName}
                onBlur={handleDateChange}
                onChange={(evt) => setFromDate(evt.target.value)}
                onKeyDown={handleKeyDownFromDate}
                min={minDate}
                max={maxDate}
                ref={fromRef}
                placeholder={DATE_VALIDATION_FORMAT}
                type="date"
                value={fromDate}
              />
              <button
                className="a-btn a-btn__link"
                onClick={() => handleClear('from')}
              >
                {iconMap.getIcon('delete')}
                <span className="u-visually-hidden">
                  Clear company received from date filter
                </span>
              </button>
            </div>
          </li>
          <li>
            <label
              className="a-label a-label__heading body-copy"
              htmlFor={`${fieldName}-through`}
            >
              Through
            </label>
            <div className="m-btn-inside-input">
              <input
                id={`${fieldName}-through`}
                className={inputThroughClassName}
                onBlur={handleDateChange}
                onChange={(evt) => setThroughDate(evt.target.value)}
                onKeyDown={handleKeyDownThroughDate}
                min={minDate}
                max={maxDate}
                placeholder={DATE_VALIDATION_FORMAT}
                ref={throughRef}
                type="date"
                value={throughDate}
              />
              <button
                className="a-btn a-btn__link"
                onClick={() => handleClear('through')}
              >
                {iconMap.getIcon('delete')}
                <span className="u-visually-hidden">
                  Clear company received through date filter
                </span>
              </button>
            </div>
          </li>
        </ul>

        {errors.length
          ? errors.map((message, key) => (
              <div key={key}>
                {message}
                <span aria-hidden="true">
                  {iconMap.getIcon('delete-round', 'cf-icon-delete-round')}
                </span>
              </div>
            ))
          : null}
      </div>
    </CollapsibleFilter>
  );
};
