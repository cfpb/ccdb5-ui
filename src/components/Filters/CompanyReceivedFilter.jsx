/* eslint complexity: ["error", 6] */
import './DateFilter.less'
import {
    DATE_VALIDATION_FORMAT,
    maxDate,
    minDate
} from '../../constants'
import React, { useEffect, useMemo, useRef, useState } from 'react'
import {
    selectQueryCompanyReceivedMax,
    selectQueryCompanyReceivedMin
} from '../../reducers/query/selectors';
import { useDispatch, useSelector } from 'react-redux'
import { changeDates } from '../../actions/filter'
import CollapsibleFilter from './CollapsibleFilter'
import dayjs from 'dayjs'
import dayjsCustomParseFormat from 'dayjs/plugin/customParseFormat'
import dayjsIsBetween from 'dayjs/plugin/isBetween'
import { formatDate } from '../../utils/formatDate';
import iconMap from '../iconMap'

dayjs.extend( dayjsCustomParseFormat )
dayjs.extend( dayjsIsBetween )

export const CompanyReceivedFilter = () => {
  const fieldName = 'company_received'
  const title = 'The date the CFPB sent the complaint to the company';
  const dateFrom = useSelector( selectQueryCompanyReceivedMin );
  const dateThrough = useSelector( selectQueryCompanyReceivedMax );
  const [ fromDate, setFromDate ] = useState( formatDate( dateFrom ) );
  const [ throughDate, setThroughDate ] = useState( formatDate( dateThrough ) );
  const dispatch = useDispatch();
  const errorMessageText = "'From' date must be less than 'through' date";

  const fromRef = useRef();
  const throughRef = useRef();

  useEffect( () => {
    // put it in YYYY-MM-DD format
    if ( dayjs( dateFrom ).isValid() ) {
      setFromDate( formatDate( dateFrom ) );
    } else {
      setFromDate( '' )
    }
    if ( dayjs( dateThrough ).isValid() ) {
      setThroughDate( formatDate( dateThrough ) );
    } else {
      setThroughDate( '' )
    }
  }, [ dateFrom, dateThrough ] );

  const handleClear = period => {
    if ( period === 'from' ) {
      dispatch( changeDates( fieldName, '', throughDate ) )
    }
    if ( period === 'through' ) {
      dispatch( changeDates( fieldName, fromDate, '' ) )
    }
  }

  const handleKeyDownFromDate = event => {
    if ( event.key === 'Enter' ) {
      fromRef.current.blur();
    }
  }

  const handleKeyDownThroughDate = event => {
    if ( event.key === 'Enter' ) {
      throughRef.current.blur();
    }
  }

  const errors = useMemo( () => {
    if ( dayjs( fromDate ).isAfter( throughDate ) ) {
      return errorMessageText;
    }
    return false;
  }, [ fromDate, throughDate ] )

  const handleDateChange = () => {
    let _throughDate = throughDate;
    let _fromDate = fromDate
    if ( !dayjs( fromDate ).isValid() ) {
      fromRef.current.value = '';
      _fromDate = ''
    }
    if ( !dayjs( throughDate ).isValid() ) {
      throughRef.current.value = '';
      _throughDate = ''
    }
    const isDateDifferent = dateFrom !== _fromDate ||
        dateThrough !== _throughDate;
    if ( isDateDifferent ) {
      dispatch( changeDates( fieldName, _fromDate, _throughDate ) )
    }
  };

  const inputStartClassName = useMemo( () => {
    const style = [ 'a-text-input' ]
    if ( dayjs( fromDate ).isBefore( minDate ) ||
        dayjs( fromDate ).isAfter( throughDate ) ) {
      style.push( 'a-text-input__error' )
    }
    return style.join( ' ' )
  }, [ fromDate, throughDate ] );

  const inputEndClassName = useMemo( () => {
    const style = [ 'a-text-input' ]
    if ( dayjs( throughDate ).isAfter( maxDate ) ||
        dayjs( throughDate ).isBefore( fromDate ) ) {
      style.push( 'a-text-input__error' )
    }
    return style.join( ' ' )
  }, [ fromDate, throughDate ] );

  return <CollapsibleFilter title={title}
                            className="aggregation date-filter">
        <div>
            <ul className="date-inputs">
                <li>
                    <label className="a-label a-label__heading body-copy"
                           htmlFor={`${ fieldName }-from`}>
                        From
                    </label>
                    <div className="m-btn-inside-input">
                        <input id={`${ fieldName }-from`}
                               className={inputStartClassName}
                               onBlur={handleDateChange}
                               onChange={evt => setFromDate( evt.target.value )}
                               onKeyDown={handleKeyDownFromDate}
                               min={minDate}
                               max={maxDate}
                               ref={fromRef}
                               placeholder={DATE_VALIDATION_FORMAT}
                               type="date"
                               value={fromDate}
                        />
                        <button className="a-btn a-btn__link"
                                onClick={() => handleClear( 'from' )}>
                            {iconMap.getIcon( 'delete' )}
                            <span className="u-visually-hidden">Clear</span>
                        </button>
                    </div>
                </li>
                <li>
                    <label className="a-label a-label__heading body-copy"
                           htmlFor={`${ fieldName }-through`}>
                        Through
                    </label>
                    <div className="m-btn-inside-input">
                        <input id={`${ fieldName }-through`}
                               className={inputEndClassName}
                               onBlur={handleDateChange}
                               onChange={
                                   evt => setThroughDate( evt.target.value )
                               }
                               onKeyDown={handleKeyDownThroughDate}
                               min={minDate}
                               max={maxDate}
                               placeholder={DATE_VALIDATION_FORMAT}
                               ref={throughRef}
                               type="date"
                               value={throughDate}
                        />
                        <button className="a-btn a-btn__link"
                                onClick={() => handleClear( 'through' )}>
                            {iconMap.getIcon( 'delete' )}
                            <span className="u-visually-hidden">Clear</span>
                        </button>
                    </div>
                </li>
            </ul>

            {errors ?
                <>{errors + ' '}
                <span aria-hidden="true">
                    {
                    iconMap.getIcon( 'delete-round', 'cf-icon-delete-round' )
                    }
                </span>
                </> :
                null
            }
        </div>

    </CollapsibleFilter>
}

