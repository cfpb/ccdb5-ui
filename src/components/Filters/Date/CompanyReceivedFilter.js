/* eslint complexity: ["error", 7] */
import './DateFilter.scss';
import { DATE_VALIDATION_FORMAT, maxDate, minDate } from '../../../constants';
import { useEffect, useMemo, useRef, useState } from 'react';
import {
  selectQueryCompanyReceivedMax,
  selectQueryCompanyReceivedMin,
} from '../../../reducers/query/selectors';
import { useDispatch, useSelector } from 'react-redux';
import { CollapsibleFilter } from '../CollapsibleFilter/CollapsibleFilter';
import dayjs from 'dayjs';
import dayjsCustomParseFormat from 'dayjs/plugin/customParseFormat';
import dayjsIsBetween from 'dayjs/plugin/isBetween';
import { formatDate } from '../../../utils/formatDate';
import getIcon from '../../Common/Icon/iconMap';
import { companyReceivedDateChanged } from '../../../reducers/query/querySlice';

dayjs.extend(dayjsCustomParseFormat);
dayjs.extend(dayjsIsBetween);

export const CompanyReceivedFilter = () => {
  const fieldName = 'company_received';
  const title = 'The date the CFPB sent the complaint to the company';
  const dateFrom = useSelector(selectQueryCompanyReceivedMin);
  const dateThrough = useSelector(selectQueryCompanyReceivedMax);
  const initialFromDate = dayjs(dateFrom).isValid() ? formatDate(dateFrom) : '';
  const initialThroughDate = dayjs(dateThrough).isValid()
    ? formatDate(dateThrough)
    : '';

  const [fromDate, setFromDate] = useState(initialFromDate);
  const [throughDate, setThroughDate] = useState(initialThroughDate);
  const dispatch = useDispatch();
  const errorMessageText = "'From' date must be less than 'through' date";

  const fromRef = useRef();
  const throughRef = useRef();

  useEffect(() => {
    // put it in YYYY-MM-DD format
    // validate to make sure it's not invalid
    const validFromDate = dateFrom ? formatDate(dateFrom) : '';
    setFromDate(validFromDate);
  }, [dateFrom]);

  useEffect(() => {
    const validThroughDate = dateThrough ? formatDate(dateThrough) : '';
    setThroughDate(validThroughDate);
  }, [dateThrough]);

  const handleKeyDownFromDate = (event) => {
    if (event.key === 'Enter') {
      fromRef.current.blur();
    }
  };

  const handleKeyDownThroughDate = (event) => {
    if (event.key === 'Enter') {
      throughRef.current.blur();
    }
  };

  const errors = useMemo(() => {
    const errs = [];
    if (dayjs(fromDate).isAfter(throughDate)) {
      errs.push(errorMessageText);
    }
    if (dayjs(fromDate).isBefore(minDate)) {
      errs.push(
        "'From' date must be after " +
          dayjs(minDate).format(DATE_VALIDATION_FORMAT),
      );
    }
    if (dayjs(throughDate).isAfter(maxDate)) {
      errs.push(
        "'Through' date must be before " +
          dayjs(maxDate).format(DATE_VALIDATION_FORMAT),
      );
    }

    return errs;
  }, [fromDate, throughDate]);

  const handleDateChange = () => {
    let _throughDate = throughDate;
    let _fromDate = fromDate;
    if (_fromDate && !dayjs(fromDate).isValid()) {
      fromRef.current.value = '';
      _fromDate = '';
    }
    if (_throughDate && !dayjs(throughDate).isValid()) {
      throughRef.current.value = '';
      _throughDate = '';
    }
    const isDateDifferent =
      dateFrom !== _fromDate || dateThrough !== _throughDate;
    if (isDateDifferent) {
      dispatch(companyReceivedDateChanged(_fromDate, _throughDate));
    }
  };

  const inputFromClassName = useMemo(() => {
    const style = ['a-text-input'];
    if (
      dayjs(fromDate).isBefore(minDate) ||
      dayjs(fromDate).isAfter(throughDate)
    ) {
      style.push('a-text-input--error');
    }
    return style.join(' ');
  }, [fromDate, throughDate]);

  const inputThroughClassName = useMemo(() => {
    const style = ['a-text-input'];
    if (
      dayjs(throughDate).isAfter(maxDate) ||
      dayjs(throughDate).isBefore(fromDate)
    ) {
      style.push('a-text-input--error');
    }
    return style.join(' ');
  }, [fromDate, throughDate]);

  return (
    <CollapsibleFilter
      title={title}
      className="aggregation date-filter"
      desc=""
    >
      <div>
        <ul className="date-inputs">
          <li>
            <label
              className="a-label a-label__heading body-copy"
              htmlFor={`${fieldName}-from`}
            >
              From
            </label>
            <div className="o-search-input">
              <div className="o-search-input__input">
                <input
                  id={`${fieldName}-from`}
                  className={inputFromClassName}
                  onBlur={handleDateChange}
                  onChange={(evt) => setFromDate(evt.target.value)}
                  onKeyDown={handleKeyDownFromDate}
                  min={minDate}
                  max={maxDate}
                  ref={fromRef}
                  placeholder={DATE_VALIDATION_FORMAT}
                  type="date"
                  value={fromDate}
                />
              </div>
            </div>
          </li>
          <li>
            <label
              className="a-label a-label__heading body-copy"
              htmlFor={`${fieldName}-through`}
            >
              Through
            </label>
            <div className="o-search-input">
              <div className="o-search-input__input">
                <input
                  id={`${fieldName}-through`}
                  className={inputThroughClassName}
                  onBlur={handleDateChange}
                  onChange={(evt) => setThroughDate(evt.target.value)}
                  onKeyDown={handleKeyDownThroughDate}
                  min={minDate}
                  max={maxDate}
                  placeholder={DATE_VALIDATION_FORMAT}
                  ref={throughRef}
                  type="date"
                  value={throughDate}
                />
              </div>
            </div>
          </li>
        </ul>

        {errors.length ? (
          <div className="a-form-alert a-form-alert--error" role="alert">
            {errors.map((message, key) => (
              <div key={key}>
                <span aria-hidden="true">
                  {getIcon('delete-round', 'cf-icon-delete-round')}
                </span>
                <div className="a-form-alert__text">{message}</div>
              </div>
            ))}
          </div>
        ) : null}
      </div>
    </CollapsibleFilter>
  );
};
