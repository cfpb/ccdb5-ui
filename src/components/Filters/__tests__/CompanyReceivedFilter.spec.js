import { CompanyReceivedFilter } from '../CompanyReceivedFilter';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import React from 'react';
import renderer from 'react-test-renderer';
import thunk from 'redux-thunk';

/**
 *
 * @param {object} query - Query state
 * @returns {Function} - Rendering function
 */
function setupSnapshot(query) {
  const middlewares = [thunk];
  const mockStore = configureMockStore(middlewares);
  const store = mockStore({
    query: query,
  });

  return renderer.create(
    <Provider store={store}>
      <CompanyReceivedFilter />
    </Provider>,
  );
}

describe('component::CompanyReceivedFilter', () => {
  describe('snapshot', () => {
    it('supports no dates in the Redux Store', () => {
      const target = setupSnapshot({});
      const tree = target.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('shows errors', () => {
      const target = setupSnapshot({
        company_received_min: new Date(2016, 0, 1),
        company_received_max: new Date(2000, 0, 1),
      });
      const tree = target.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
