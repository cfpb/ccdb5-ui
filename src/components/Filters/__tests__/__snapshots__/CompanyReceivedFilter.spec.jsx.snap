// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component::CompanyReceivedFilter snapshot shows errors 1`] = `
<section
  className="o-expandable aggregation date-filter"
>
  <div
    className="o-expandable_header o-expandable_target"
  >
    <h4
      className="o-expandable_header-left o-expandable_label"
    >
      The date the CFPB sent the complaint to the company
    </h4>
    <span
      className="o-expandable_header-right o-expandable_link"
    >
      <button
        aria-label="Hide The date the CFPB sent the complaint to the company filter"
        className="a-btn a-btn__link o-expandable_cue o-expandable_cue-close"
        onClick={[Function]}
      >
        Hide
        <svg
          className="cf-icon-svg"
          viewBox="0 0 1000 1200"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M500 105.2c-276.1 0-500 223.9-500 500s223.9 500 500 500 500-223.9 500-500-223.9-500-500-500zm263.1 550.7H236c-27.6 0-50-22.4-50-50s22.4-50 50-50h527.1c27.6 0 50 22.4 50 50s-22.4 50-50 50z"
          />
        </svg>
      </button>
    </span>
  </div>
  <p />
  <div>
    <ul
      className="date-inputs"
    >
      <li>
        <label
          className="a-label a-label__heading body-copy"
          htmlFor="company_received-from"
        >
          From
        </label>
        <div
          className="m-btn-inside-input"
        >
          <input
            className="a-text-input a-text-input__error"
            id="company_received-from"
            max="2020-05-05"
            min="2011-12-01"
            onBlur={[Function]}
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="M/D/YYYY"
            type="date"
            value="2016-01-01"
          />
          <button
            className="a-btn a-btn__link"
            onClick={[Function]}
          >
            <svg
              className="cf-icon-svg"
              viewBox="0 0 718.9 1200"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M451.4 613.7l248.1-248.1c25.6-25.1 26-66.3.8-91.9s-66.3-26-91.9-.8l-.8.8-248.1 248.1-248.1-248.1c-25.4-25.4-66.5-25.4-91.9 0s-25.4 66.5 0 91.9l248.1 248.1L19.5 861.8c-25.6 25.1-26 66.3-.8 91.9s66.3 26 91.9.8l.8-.8 248.1-248.1 248.1 248.1c25.4 25.4 66.5 25.4 91.9 0s25.4-66.5 0-91.9L451.4 613.7z"
              />
            </svg>
            <span
              className="u-visually-hidden"
            >
              Clear
            </span>
          </button>
        </div>
      </li>
      <li>
        <label
          className="a-label a-label__heading body-copy"
          htmlFor="company_received-through"
        >
          Through
        </label>
        <div
          className="m-btn-inside-input"
        >
          <input
            className="a-text-input a-text-input__error"
            id="company_received-through"
            max="2020-05-05"
            min="2011-12-01"
            onBlur={[Function]}
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="M/D/YYYY"
            type="date"
            value="2000-01-01"
          />
          <button
            className="a-btn a-btn__link"
            onClick={[Function]}
          >
            <svg
              className="cf-icon-svg"
              viewBox="0 0 718.9 1200"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M451.4 613.7l248.1-248.1c25.6-25.1 26-66.3.8-91.9s-66.3-26-91.9-.8l-.8.8-248.1 248.1-248.1-248.1c-25.4-25.4-66.5-25.4-91.9 0s-25.4 66.5 0 91.9l248.1 248.1L19.5 861.8c-25.6 25.1-26 66.3-.8 91.9s66.3 26 91.9.8l.8-.8 248.1-248.1 248.1 248.1c25.4 25.4 66.5 25.4 91.9 0s25.4-66.5 0-91.9L451.4 613.7z"
              />
            </svg>
            <span
              className="u-visually-hidden"
            >
              Clear
            </span>
          </button>
        </div>
      </li>
    </ul>
    <div>
      'From' date must be less than 'through' date
      <span
        aria-hidden="true"
      >
        <svg
          className="cf-icon-svg cf-icon-delete-round"
          viewBox="0 0 1000 1200"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M500 105.2c-276.1 0-500 223.9-500 500s223.9 500 500 500 500-223.9 500-500-223.9-500-500-500zm261.8 692.2c19.4 19.6 19.3 51.3-.3 70.7-19.5 19.3-50.9 19.3-70.4 0L499.6 676.6 308 868.1c-19.6 19.4-51.3 19.3-70.7-.3-19.3-19.5-19.3-50.9 0-70.4l191.6-191.5-191.6-191.6c-19.3-19.8-18.9-51.4.9-70.7 19.4-18.9 50.4-18.9 69.8 0l191.6 191.5 191.5-191.5c19.6-19.4 51.3-19.3 70.7.3 19.3 19.5 19.3 50.9 0 70.4L570.3 605.9l191.5 191.5z"
          />
        </svg>
      </span>
    </div>
  </div>
</section>
`;

exports[`component::CompanyReceivedFilter snapshot supports no dates in the Redux Store 1`] = `
<section
  className="o-expandable aggregation date-filter"
>
  <div
    className="o-expandable_header o-expandable_target"
  >
    <h4
      className="o-expandable_header-left o-expandable_label"
    >
      The date the CFPB sent the complaint to the company
    </h4>
    <span
      className="o-expandable_header-right o-expandable_link"
    >
      <button
        aria-label="Hide The date the CFPB sent the complaint to the company filter"
        className="a-btn a-btn__link o-expandable_cue o-expandable_cue-close"
        onClick={[Function]}
      >
        Hide
        <svg
          className="cf-icon-svg"
          viewBox="0 0 1000 1200"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M500 105.2c-276.1 0-500 223.9-500 500s223.9 500 500 500 500-223.9 500-500-223.9-500-500-500zm263.1 550.7H236c-27.6 0-50-22.4-50-50s22.4-50 50-50h527.1c27.6 0 50 22.4 50 50s-22.4 50-50 50z"
          />
        </svg>
      </button>
    </span>
  </div>
  <p />
  <div>
    <ul
      className="date-inputs"
    >
      <li>
        <label
          className="a-label a-label__heading body-copy"
          htmlFor="company_received-from"
        >
          From
        </label>
        <div
          className="m-btn-inside-input"
        >
          <input
            className="a-text-input"
            id="company_received-from"
            max="2020-05-05"
            min="2011-12-01"
            onBlur={[Function]}
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="M/D/YYYY"
            type="date"
            value="Invalid Date"
          />
          <button
            className="a-btn a-btn__link"
            onClick={[Function]}
          >
            <svg
              className="cf-icon-svg"
              viewBox="0 0 718.9 1200"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M451.4 613.7l248.1-248.1c25.6-25.1 26-66.3.8-91.9s-66.3-26-91.9-.8l-.8.8-248.1 248.1-248.1-248.1c-25.4-25.4-66.5-25.4-91.9 0s-25.4 66.5 0 91.9l248.1 248.1L19.5 861.8c-25.6 25.1-26 66.3-.8 91.9s66.3 26 91.9.8l.8-.8 248.1-248.1 248.1 248.1c25.4 25.4 66.5 25.4 91.9 0s25.4-66.5 0-91.9L451.4 613.7z"
              />
            </svg>
            <span
              className="u-visually-hidden"
            >
              Clear
            </span>
          </button>
        </div>
      </li>
      <li>
        <label
          className="a-label a-label__heading body-copy"
          htmlFor="company_received-through"
        >
          Through
        </label>
        <div
          className="m-btn-inside-input"
        >
          <input
            className="a-text-input"
            id="company_received-through"
            max="2020-05-05"
            min="2011-12-01"
            onBlur={[Function]}
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="M/D/YYYY"
            type="date"
            value="Invalid Date"
          />
          <button
            className="a-btn a-btn__link"
            onClick={[Function]}
          >
            <svg
              className="cf-icon-svg"
              viewBox="0 0 718.9 1200"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M451.4 613.7l248.1-248.1c25.6-25.1 26-66.3.8-91.9s-66.3-26-91.9-.8l-.8.8-248.1 248.1-248.1-248.1c-25.4-25.4-66.5-25.4-91.9 0s-25.4 66.5 0 91.9l248.1 248.1L19.5 861.8c-25.6 25.1-26 66.3-.8 91.9s66.3 26 91.9.8l.8-.8 248.1-248.1 248.1 248.1c25.4 25.4 66.5 25.4 91.9 0s25.4-66.5 0-91.9L451.4 613.7z"
              />
            </svg>
            <span
              className="u-visually-hidden"
            >
              Clear
            </span>
          </button>
        </div>
      </li>
    </ul>
  </div>
</section>
`;
